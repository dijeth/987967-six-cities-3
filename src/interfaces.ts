import { id, roomType, coord } from "./types";

export interface OfferMini {
    readonly id: id;
    title: string;
    type: roomType;
    pictures: Array<string>;
    cost: number;
    rating: number;
    isPremium: boolean;
    isFavorite: boolean;
    city: string;
    coord: coord;
};

export interface Offer extends OfferMini {
    bedroomCount: number;
    adultsCount: number;
    insideFeatures: Array<string>;
    userName: string;
    userPicture: string;
    isSuperUser: boolean;
    descriptionTitle: string;
    description: string;
    zoom: number;
}

export interface City {
    name: string;
    zoom: number;
    centerCoord: coord;
}

export interface UserReview {
    readonly id: id;
    userName: string;
    userPicture: string;
    rating: number;
    description: string;
    date: string;
}