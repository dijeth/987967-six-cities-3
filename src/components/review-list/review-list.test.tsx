import * as React from 'react';
import * as renderer from 'react-test-renderer';
import ReviewList from './review-list';
import {UserReview} from '../../interfaces';

const mocks: Array<UserReview> = [{
  id: `id-1`,
  userName: `userName-1`,
  userPicture: `userPicture-1`,
  rating: 4.8,
  description: `description-1`,
  date: new Date(1582194976548).toISOString()
},
{
  id: `id-2`,
  userName: `userName-2`,
  userPicture: `userPicture-2`,
  rating: 3.8,
  description: `description-2`,
  date: new Date(1583000000000).toISOString()
},
];

it(`<ReviewList /> should be renderer correctly`, () => {
  const reviewList = renderer.create(<ReviewList reviews={mocks} />).toJSON();
  expect(reviewList).toMatchSnapshot();
});
