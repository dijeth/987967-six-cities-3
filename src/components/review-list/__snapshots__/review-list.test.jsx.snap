// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewList /> should be renderer correctly 1`] = `
Array [
  <h2
    className="reviews__title"
  >
    Reviews Â· 
    <span
      className="reviews__amount"
    >
      2
    </span>
  </h2>,
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="userPicture-2"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          userName-2
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          description-2
        </p>
        <time
          className="reviews__time"
          dateTime="2020-02-29"
        >
          February 2020
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="userPicture-1"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          userName-1
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          description-1
        </p>
        <time
          className="reviews__time"
          dateTime="2020-02-20"
        >
          February 2020
        </time>
      </div>
    </li>
  </ul>,
]
`;
